### DISCUSSION TITLE
Gathering all ICE candidates into one SDP response

### DISCUSSION THREAD

**Discussion by tristan-f-r** [webrtc-rs/webrtc#218] [Q&A] [answered]
How can I gather all the ice candidates before sending the SDP data?

**Comment by k0nserv** [comment, marked as answer]
Many of the examples do this via `gathering_complete_promise`, .e.g [the reflect example](https://github.com/webrtc-rs/examples/blob/5a0e2861c66a45fca93aadf9e70a5b045b26dc9e/examples/reflect/reflect.rs#L296)

```rust
peer_connection.gathering_complete_promise().await
```

Keep in mind that you have to call `create_answer` before awaiting this future because candidate gathering only starts at that point.
